File: Kers/ClientApp/src/app/modules/reporting/modules/training/training.ts
import {ExtensionEvent} from '../events/extension-event';
import { User, PlanningUnit } from '../user/user.service';

export class Training extends ExtensionEvent{
    classicInServiceTrainingId:number;
    submittedBy: User;
    submittedById:number;
    approvedBy:User;
    approvedDate?:Date;
    tID:string;
    tStatus:string;
    sessionCancelledDate?:Date;
    trainDateBegin:string;
    trainDateEnd:string;
    registerCutoffDaysId?:number;
    registerCutoffDays:TainingRegisterWindow;
    cancelCutoffDaysId?:number;
    cancelCutoffDays: TrainingCancelEnrollmentWindow;
    iHourId?:number;
    iHour:TainingInstructionalHour;
    seatLimit?:number;
    day1:string;
    day2:string;
    day3:string;
    day4:string;
    tContact:string;
    tAudience:string;
    enrollment:TrainingEnrollment[];
    qualtricsSurveyID:string;
    evaluationLink:string;
    createdDateTime:Date;
    sessions: TrainingSession[];
    trainingSession: TrainingSession[];
    trainingSessionWithTimes:TrainingSession[];
    surveyResults:TrainingSurveyResult[];
    etimezone:boolean;
}
export class TrainingSurveyResult{
    id:number;
    result:string;
    userId:number;
    user:User;
    training:Training;
    trainingId:number;
    created:Date;
}

export class TainingRegisterWindow{
    id:number;
    registerDaysVal:string;
    registerDaysTxt:string;

}

export class TrainingCancelEnrollmentWindow{
    id:number;
    cancelDaysVal: number;
    cancelDaysTxt:string;
}

export class TainingInstructionalHour{
    id:number;
    iHoursTxt:string;
    iHourValue:number;
}

export class TrainingEnrollment{
    id:number;
    rDT?:Date;
    puid:string;
    PlanningUnitId:number;
    planningUnit:PlanningUnit;
    attendie:User;
    attendieId:number;
    trainingId:string;
    training: Training;
    eStatus:string;
    enrolledDate:Date;
    cancelledDate:Date;
    attended?:boolean;
    evaluationMessageSent?:boolean;
}

export class TrainingSearchCriteria{
    start: string;
    end: string;
    search: string = "";
    status: string;
    contacts: string = "";
    day?: number;
    order: string = 'dsc';
    withseats: boolean = false;
    attendance: boolean = false;
    admin: boolean = false;
  }

  export class TrainingSession{
      id:number;
      date: Date | Object;
      starttime:string;
      endtime:string;
      note:string;
      index:number;
      start:string;
      end:string;
  }

File: KersDb/Entities/KERScore/Training.cs
namespace Kers.Models.Entities.KERScore
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.IO;
    using System.Runtime.Serialization;
    using Newtonsoft.Json;
    public class Training : ExtensionEvent
    {
        public int classicInServiceTrainingId { get; set; }
        public KersUser submittedBy {get;set;}
        public int? submittedById {get;set;}
        public KersUser approvedBy {get;set;}
        public DateTime? approvedDate {get;set;}

        [StringLength(25)]
        public string tID { get; set; }

        [StringLength(25)]
        public string tStatus { get; set; }

        public DateTime? sessionCancelledDate { get; set; }

        [StringLength(8)]
        public string TrainDateBegin { get; set; }

        [StringLength(8)]
        public string TrainDateEnd { get; set; }

        public int? RegisterCutoffDaysId { get; set; }
        public TainingRegisterWindow RegisterCutoffDays { get; set; }

        public int? CancelCutoffDaysId { get; set; }
        public TrainingCancelEnrollmentWindow CancelCutoffDays { get; set; }

        public int? iHourId { get; set; }
        public TainingInstructionalHour iHour { get; set; }

        public int? seatLimit { get; set; }

        [StringLength(300)]
        public string tTime { get; set; }

        [StringLength(300)]
        public string day1 { get; set; }

        [StringLength(300)]
        public string day2 { get; set; }

        [StringLength(300)]
        public string day3 { get; set; }

        [StringLength(300)]
        public string day4 { get; set; }

        public string tAudience { get; set; }
        public List<TrainingEnrollment> Enrollment {get; set; }
        public List<TrainingSession> TrainingSession {get;set;}

        [StringLength(200)]
        public string qualtricsSurveyID { get; set; }

        public string evaluationLink { get; set; }
        //public List<TrainingSurveyResult> SurveyResults {get;set;}


    }
}